backend/server.js

```diff
@@ -67,8 +72,43 @@ io.on("connection", (socket) => {
     });
   });
 
-  socket.off("setup", () => {
-    console.log("USER DISCONNECTED");
+  // Event for reacting to a message
+socket.on("reactMessage", async ({ messageId, userId, reaction }) => {
+  try {
+    // Find the message by ID
+    const message = await Message.findById(messageId);
+    
+    if (!message) {
+      console.error("Message not found");
+      return;
+    }
+
+    // Check if the user has already reacted
+    const existingReactionIndex = message.reactions.findIndex(
```

Comment:
can we add a unique constrain in mongodb? that way, we don't have to check this.